homeassistant/components/amcrest/camera.py

```diff
@@ -139,22 +167,32 @@ def is_on(self):
 
     # Other Entity method overrides
 
+    async def async_added_to_hass(self):
+        """Add camera to list."""
+        self.hass.data[DATA_AMCREST]['cameras'].append(self)
```

Comment:
We're not allowed to store entities in shared containers. Why do we need to store it?

Reply:
The use case are the service handlers. Use our dispatch helper instead. Connect the signal here to the correct entity methods.

Reply:
@MartinHjelmare we should probably open a discussion if we should keep enforcing this, or that it's up to the integrations author if they want a decoupled design.

Reply:
I've been thinking a bit about that. But so far I've only seen potential problems when not using dispatcher.

It's good to keep the platform contained and not expose it. Refactoring is easier. It's the same argument why we shouldn't interact directly with entities in tests but use the core to set up and assert.

It's also easier to introduce a bug and call a method before the entity has been added, when not using dispatcher, which often isn't allowed. With the dispatcher we've always follow the same pattern and use `async_added_to_hass`.